pkg/providers/instancetype/instancetype.go

```diff
@@ -247,6 +261,52 @@ func (p *DefaultProvider) UpdateInstanceTypeOfferings(ctx context.Context) error
 	return nil
 }
 
+func (p *DefaultProvider) UpdateVMCapacityCache(ctx context.Context, kubeClient client.Client, nodeName string) error {
+
+	node := &corev1.Node{}
+	if err := kubeClient.Get(ctx, client.ObjectKey{Name: nodeName}, node); err != nil {
+		return fmt.Errorf("failed to get node: %w", err)
+	}
+
+	nodeClaimList := &karpv1.NodeClaimList{}
+	if err := kubeClient.List(ctx, nodeClaimList); err != nil {
+		return fmt.Errorf("failed to list nodeclaims: %w", err)
+	}
+	nodeClaim, found := lo.Find(nodeClaimList.Items, func(nc karpv1.NodeClaim) bool {
+		return nc.Status.NodeName == node.Name
+	})
+	if !found {
+		return fmt.Errorf("failed to find nodeclaim")
+	}
+
+	nodeClass := &v1.EC2NodeClass{}
+	if err := kubeClient.Get(ctx, client.ObjectKey{Name: nodeClaim.Spec.NodeClassRef.Name}, nodeClass); err != nil {
+		return fmt.Errorf("failed to get ec2nodeclass: %w", err)
+	}
+
+	// Ensure AMI is current
+	if !lo.ContainsBy(nodeClass.Status.AMIs, func(ami v1.AMI) bool {
+		return ami.ID == nodeClaim.Status.ImageID
+	}) {
+		return nil
+	}
```

Comment:
Don't remove it, I definitely think it should be included. And it should work today, using `MapToInstanceTypes` basically just ensures it's future proof.

Reply:
@jmdeal could you please explain this a bit more? In order to call MapToInstanceTypes I'd need to also setup the related cloudprovider.InstanceType (unless I'm missing that being available somewhere) which seems a bit more involved than necessary.
My code doesn't consider the AMI requirements and simply checks whether the NodeClaim's AMI exists under the NodeClass and if not, skips it. If additional flexibility comes to AMI requirements wouldn't that just mean more AMIs under nodeClass.Status.AMIs?

Reply:
Added the below but would like to understand why this is preferable to comparing the NodeClaim ImageID to nodeClass.Status.AMIs.
```
	// Ensure AMI is current
	instanceTypes, err := p.List(ctx, nodeClass)
	amiMap := amifamily.MapToInstanceTypes(instanceTypes, nodeClass.Status.AMIs)
	compatible := amiMap[nodeClaim.Status.ImageID]
	instanceType, found := lo.Find(compatible, func(i *cloudprovider.InstanceType) bool {
		return i.Name == node.Labels[corev1.LabelInstanceTypeStable]
	})
	if !found {
		return nil
	}
```

Reply:
@jmdeal could you please explain this a bit more? If additional flexibility comes to AMI requirements wouldn't that just mean more AMIs under nodeClass.Status.AMIs?

Added the below to implement MapToInstanceTypes:
```
	// Ensure AMI is current
	instanceTypes, err := p.List(ctx, nodeClass)
	amiMap := amifamily.MapToInstanceTypes(instanceTypes, nodeClass.Status.AMIs)
	compatible := amiMap[nodeClaim.Status.ImageID]
	instanceType, found := lo.Find(compatible, func(i *cloudprovider.InstanceType) bool {
		return i.Name == node.Labels[corev1.LabelInstanceTypeStable]
	})
	if !found {
		return nil
	}
```

Reply:
Calling List will make things a bit worse without including something like `discoveredCapacityCacheSeqNum` as it will trigger updating instanceTypesCache. Was there a different way you were suggesting to handle this?